app/controllers/course/assessment/submissions_controller.rb

```diff
@@ -108,17 +56,18 @@ def authorize_submission!
     end
   end
 
-  def load_or_create_answers
-    return unless @submission.attempting?
+  def load_service
```

Comment:
What can be done also is that the controller just acts as a dispatcher, so instead of using an around callback, you just have something that returns the correct service object type, then update looks like:

```ruby
def update
  service = service_class.new(self, assessment: assessment, submission: submission)
  service.update
  extract_instance_variables(service)
end
```

So the service does not know about the controller.